packages/nextra-theme-docs/src/components/search.tsx

```diff
@@ -213,113 +121,105 @@ export function Search({
     emitted: false
   })
   const handleChange = useCallback(
-    (e: ChangeEvent<HTMLInputElement> | CompositionEvent<HTMLInputElement>) => {
+    (event: SyntheticEvent<HTMLInputElement>) => {
       if (!compositionStateRef.current.compositioning) {
-        const { value } = e.currentTarget
-        onChangeProp(value)
-        setShow(Boolean(value))
+        const { value } = event.currentTarget
+        onChange(value)
         compositionStateRef.current.emitted = true
         return
       }
       compositionStateRef.current.emitted = false
     },
-    [onChangeProp]
+    [onChange]
   )
```

Comment:
is also related to composition but is no longer needed at the moment.

remove `compositionStateRef` and change to 

```ts
const handleChange = useCallback(
  (event: SyntheticEvent<HTMLInputElement>) => {
      const { value } = event.currentTarget
      onChange(value)
  },
  [onChange]
)
```

Reply:
removed here https://github.com/shuding/nextra/pull/3186/commits/5809fe9b355682b506cd2cf93ff84a270f00ee41 and did cleanup of a select component to avoid using render props

also, forget to mention that on mobile search results now take full height, i think is better

https://github.com/user-attachments/assets/4bdc3914-c3de-4c29-86a2-8ae49d7cb199